prog/victoria_metrics/victoria_metrics_resource_nexus.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::VictoriaMetrics::VictoriaMetricsResourceNexus < Prog::Base
+  subject_is :victoria_metrics_resource
+
+  def self.assemble(project_id, name, location_id, admin_user, admin_password, storage_size_gib, node_count, vm_size)
+    project = Project[project_id]
+    fail "No existing project" unless project
+
+    location = Location[location_id]
+    fail "No existing location" unless location
+
+    unless node_count == 1
+      fail "Only single node is supported"
+    end
+
+    Validation.validate_vm_size(vm_size, "x64")
+    Validation.validate_name(name)
+    Validation.validate_minio_username(admin_user)
```

Comment:
`validate_victoria_metrics_username` or if you want to re-use something, maybe `validate_name`?